test/lib/yggdrasil/player/registry_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule PlayerRegistryTests do
+  use ExUnit.Case, async: true
+  alias Yggdrasil.Player.Registry
+
+  setup do
+    {:ok, registry} = Registry.start_link
+    {:ok, registry: registry}
+  end
+
+  test "it allows a player to registry as online" do
+    user_id = 1
+    assert :ok = Registry.register_player user_id, self
+  end
+
+  test "it knows if a player is online" do
+    user_id = 1
+    refute Registry.is_online user_id
+    assert :ok = Registry.register_player user_id, self
+    assert Registry.is_online user_id
+  end
+
+  test "it will not allow the same player to register twice" do
+    user_id = 1
+
+    assert :ok = Registry.register_player user_id, self
+
+    register_task = Task.async(fn -> Registry.register_player user_id, self end)
+    assert {:error, :already_registered} = Task.await(register_task)
+  end
+
+  test "it knows when a player exits" do
+    user_id = 1
+
+    register_task = Task.async(fn -> Registry.register_player user_id, self end)
+
+    # verify player was registered successfully
+    assert :ok = Task.await(register_task)
+
+    refute Registry.is_online user_id
```

Comment:
Yeah noticed that as well in the cleanup of the ets, you are using state[monitor] as the key.

I'm unsure what you are testing here.

In this test the following seems to happen

* setup runs starts a new registry (no players added)
* you register player 1
* you assert that the player was registered 
* you __refute__ that the user_id is_online

I would think that refute would always fail in this case right?